sklearn/feature_selection/sequential_feature_selector.py

```diff
@@ -0,0 +1,332 @@
+        orig_set = set(range(self._n_features))
+        if self.forward:
+            if select_in_range:
+                k_to_select = self.n_features_to_select[1]
+            k_idx = ()
+            k = 0
+        else:
+            if select_in_range:
+                k_to_select = self.n_features_to_select[0]
+            k_idx = tuple(range(self._n_features))
+            k = len(k_idx)
+            k_score = self._calc_score(X, y, k_idx, cloned_estimator)
+            self.subsets_[k] = {
+                'feature_subset_idx': k_idx,
+                'cv_scores': k_score,
+                'avg_score': np.mean(k_score)
+                }
+
+        best_subset = None
+        k_score = 0
+
+        while k != k_to_select:
+            prev_subset = set(k_idx)
+            if self.forward:
+                k_idx, k_score, cv_scores = self._inclusion(
+                    orig_set=orig_set,
+                    subset=prev_subset,
+                    X=X,
+                    y=y,
+                    estimator=cloned_estimator
+                )
+            else:
+                k_idx, k_score, cv_scores = self._exclusion(
+                    feature_set=prev_subset,
+                    X=X,
+                    y=y,
+                    estimator=cloned_estimator
+                )
+
+            k = len(k_idx)
+            if k not in self.subsets_ or (self.subsets_[k]['avg_score'] <
+                                          k_score):
+                self.subsets_[k] = {
+                    'feature_subset_idx': k_idx,
+                    'cv_scores': cv_scores,
+                    'avg_score': k_score
+                }
+
+        if select_in_range:
+            max_score = float('-inf')
+            for k in self.subsets_:
+                if (k < self.n_features_to_select[0] or
+                        k > self.n_features_to_select[1]):
+                    continue
+                if self.subsets_[k]['avg_score'] > max_score:
+                    max_score = self.subsets_[k]['avg_score']
+                    best_subset = k
+            k_score = max_score
+            k_idx = self.subsets_[best_subset]['feature_subset_idx']
+
+        self.support_ = k_idx
+        self.support_ = self._get_support_mask()
+        self.score_ = k_score
+        return self
+
+    def _calc_score(self, X, y, indices, estimator):
+        if self.cv:
+            scores = cross_val_score(estimator,
+                                     X[:, indices], y,
+                                     cv=self.cv,
+                                     scoring=self._scorer,
+                                     n_jobs=self.n_jobs,
+                                     pre_dispatch=self.pre_dispatch)
+        else:
+            estimator.fit(X[:, indices], y)
+            scores = np.array([self._scorer(estimator,
+                               X[:, indices], y)])
+        return scores
+
+    def _inclusion(self, orig_set, subset, X, y, estimator):
```

Comment:
can we refactor make this and exclusion just generators of feature subsets? I.e.
```py
    def _generate_inclusion(self, orig_set, subset):
        remaining = orig_set - subset
        if remaining:
            for feature in remaining:
                yield tuple(subset | {feature})

    def _generate_exclusion(self, feature_set, fixed_feature=None):
        n = len(feature_set)
        if n > 1:
            for p in combinations(feature_set, r=n - 1):
                if fixed_feature and fixed_feature not in set(p):
                    continue
                yield p
```

Reply:
will put that onto the todo list (once all unit tests are in place)